Source/EventHorizon/Consumer/Subscriptions.cs

```diff
@@ -17,18 +19,21 @@ namespace Dolittle.Runtime.EventHorizon.Consumer
     public class Subscriptions : ISubscriptions
     {
         readonly ConcurrentDictionary<SubscriptionId, Subscription> _subscriptions = new ConcurrentDictionary<SubscriptionId, Subscription>();
-        readonly IStreamProcessorStateRepository _streamProcessorStates;
+        readonly IResilientStreamProcessorStateRepository _streamProcessorStates;
+        readonly IPolicies _policies;
         readonly ILoggerManager _loggerManager;
         readonly ILogger<Subscriptions> _logger;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Subscriptions"/> class.
         /// </summary>
-        /// <param name="streamProcessorStates">The <see cref="IStreamProcessorStateRepository" />.</param>
+        /// <param name="streamProcessorStates">The <see cref="IResilientStreamProcessorStateRepository" />.</param>
+        /// <param name="policies">The <see cref="IPolicies" />.</param>
         /// <param name="loggerManager">The <see cref="ILoggerManager" />.</param>
-        public Subscriptions(IStreamProcessorStateRepository streamProcessorStates, ILoggerManager loggerManager)
+        public Subscriptions(IResilientStreamProcessorStateRepository streamProcessorStates, IPolicies policies, ILoggerManager loggerManager)
```

Comment:
Why not `IAsyncPolicyFor<ICanFetchEventsFromStream>` ?

Reply:
https://github.com/dolittle-runtime/Runtime/pull/408/commits/554add08e459ccc031019dcbde7d23b4f6928a23